cmd/fluxctl/completion_cmd.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	completionShells = map[string]func(out io.Writer, cmd *cobra.Command) error{
+		"bash": runCompletionBash,
+		"zsh":  runCompletionZsh,
+	}
+)
+
+func newCompletionCommand() *cobra.Command {
+	shells := []string{}
+	for s := range completionShells {
+		shells = append(shells, s)
+	}
+
+	return &cobra.Command{
+		Use:                   "completion shell",
```

Comment:
Do we want to distinguish between `completion` which is literal, and `shell` which is user input? (or indeed, give the possible values)

Reply:
I think it would make sense here to utilize the `shells` (again), and put `<>` around the list to highlight the fact that they are options.

Reply:
Good idea, only issue is that the `Use` field is also used by the root command as a way to distinguish which of the commands need to skip port forward. So the completion command would have to get instantiated there (or come up with a different way of detecting which commands are run). Alternatively, we could change it to `completion SHELL` to distinguish them, similar to `kubectl`.  What do you think?

Reply:
Given people are hopefully already familiar with `kubectl`, I think that’s a good alternative.